### PR TITLE
Implement GetAppProfile() and DeleteAppProfile().

### PR DESCRIPTION
**Pull request by coryan**
This is part of the fixes for #780. It implements GetAppProfile(), DeleteAppProfile(),
the unit tests for each, the samples for each, runs the examples as part of the CI
build, and extends the integration test to use both functions.

### CODE REVIEW

File: google/cloud/bigtable/examples/run_examples_emulator.sh [hunk 1]
```diff
@@ -31,16 +31,17 @@ readonly ZONE_ID="fake-zone"
 
 # The examples are noisy,
 for example in instance_admin table_admin data; do
-  LOG=$(mktemp --tmpdir "bigtable_examples_${example}_XXXXXXXXXX.log")
+  log=$(mktemp --tmpdir "bigtable_examples_${example}_XXXXXXXXXX.log")
   echo "${COLOR_GREEN}[ RUN      ]${COLOR_RESET} ${example}"
-  run_all_${example}_examples "${PROJECT_ID}" "${ZONE_ID}" >${LOG} 2>&1 </dev/null
+  run_all_${example}_examples "${PROJECT_ID}" "${ZONE_ID}" # >${log} 2>&1 </dev/null
```
Comment (mbrukman) [inline google/cloud/bigtable/examples/run_examples_emulator.sh]: This is a new comment character; you are losing all logs here. Was this change intended?

File: google/cloud/bigtable/internal/instance_admin_test.cc [hunk 2]
```diff
@@ -591,8 +649,10 @@ auto create_list_app_profiles_lambda =
         EXPECT_NE(nullptr, response);
         for (auto const& app_profile_id : app_profile_ids) {
           auto& app_profile = *response->add_app_profiles();
-          app_profile.set_name(instance_name + "/appProfiles/" +
-                               app_profile_id);
+          auto name = instance_name;
```
Comment (mbrukman) [inline google/cloud/bigtable/internal/instance_admin_test.cc]: Combine into one line? Or are the `+=` appends more efficient?

Comment (coryan) [reply google/cloud/bigtable/internal/instance_admin_test.cc]: According to clang-tidy (on my IDE, I do not believe we have turned that on for the CI builds) they are.  I do not know this to be true for a fact, but keeping clang-tidy happy seems worthwhile.

File: google/cloud/bigtable/examples/run_examples_emulator.sh (hunk 1 above)
Comment (coryan) [reply google/cloud/bigtable/examples/run_examples_emulator.sh]: Nope, that was a mistake. Fixed, and thanks!

**Comment by coryan** [comment]
PTAL.

File: google/cloud/bigtable/examples/bigtable_samples_instance_admin.cc [hunk 3]
```diff
@@ -294,17 +309,73 @@ void ListAppProfiles(google::cloud::bigtable::InstanceAdmin instance_admin,
...
+  bool ignore_warnings = true;
+  if (argc >= 2) {
+    std::string arg = ConsumeArg(argc, argv);
+    if (arg == "true") {
+      ignore_warnings = true;
+    } else if (arg == "false") {
+      ignore_warnings = false;
+    } else {
+      throw Usage{
+          "delete-app-profile: ignore-warnings parameter must be either"
```
Comment (mbrukman) [inline google/cloud/bigtable/examples/bigtable_samples_instance_admin.cc]: Maybe this should just print the same usage messages as above, with the complete set of arguments? Just a thought — if this error message is shown, it's unclear *which* argument is the `ignore-warnings` in this case.

File: google/cloud/bigtable/internal/instance_admin_test.cc (hunk 2 above)
Comment (mbrukman) [reply google/cloud/bigtable/internal/instance_admin_test.cc]: Interesting; I see the concatenations with `+` on a single line elsewhere, but multiple lines with `+=` here; would be curious some day to find out why they're treated differently. Not suggesting any action items for you; just thinking out loud.

File: google/cloud/bigtable/examples/bigtable_samples_instance_admin.cc (hunk 3 above)
Comment (coryan) [reply google/cloud/bigtable/examples/bigtable_samples_instance_admin.cc]: Yup.  Fixed.
